src/site/_utils/map-value.js

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Returns value from map, or default if no value found.
+ *
+ * @param {Map<string, any>} map Map to get value from.
+ * @param {string} key Key to get value from map.
+ * @param {any} def Default value to return.
+ * @return {any} Value found, or default if no value returned.
+ */
+module.exports = (map, key, def = []) => {
```

Comment:
FWIW this is a good idea but Python for example has [.setdefault](https://docs.python.org/2/library/stdtypes.html#dict.setdefault) which is basically more convenient.

```js
setDefault(map, foo, []).push(123);
```